package/MDAnalysis/lib/pkdtree.py

```diff
@@ -162,53 +164,55 @@ def find_centers(self, center_point, radius):
-            # distance to the plane containing the origin
-            if np.dot(wrapped_c, self._rm[i]) < extents[i]:
-                displacements.append(self._dm[i])  # "upper" image
-            # distance to the plane containing point self._dm[i]
-            elif np.dot(self._dm[i] - wrapped_c, self._rm[i]) < extents[i]:
-                displacements.append(-self._dm[i])  # "lower" image
-        # If we have displacements along more than one axis, we have
-        # to combine them. This happens when wrapped_center is close
-        # to any edge or vertex of the central cell.
-        # face, n_displacements==1; no combination
-        # edge, n_displacements==2; combinations produce one extra displacement
-        # vertex, n_displacements==3; five extra displacements
-        n_displacements = len(displacements)
-        if n_displacements > 1:
-            for start in range(n_displacements - 1, -1, -1):
-                for i in range(start+1, len(displacements)):
-                    displacements.append(displacements[start]+displacements[i])
-        return [wrapped_c, ] + [wrapped_c + d for d in displacements]
-
-    def search(self, center, radius):
-        """Search all points within radius of center and its periodic images.
+        extents = np.where(self._extents < radius, self._extents, radius)
+        images = list()
+        for center_point in center_points:
+            # displacements are vectors that we add to center_point to
+            # generate images "up" or "down" the central cell along
+            # the axis that we happen to be looking.
+            displacements = list()
+            for i in range(self.dim):
+                # distance to the plane containing the origin
+                if np.dot(center_point, self._rm[i]) < extents[i]:
+                    displacements.append(self._dm[i])  # "upper" image
+                # distance to the plane containing point self._dm[i]
+                elif np.dot(self._dm[i] - center_point, self._rm[i]) < extents[i]:
+                    displacements.append(-self._dm[i])  # "lower" image
+            # If we have displacements along more than one axis, we have
+            # to combine them. This happens when center_point is close
+            # to any edge or vertex of the central cell.
+            # face, n_displacements==1; no combination
+            # edge, n_displacements==2; combinations produce one extra displacement
+            # vertex, n_displacements==3; five extra displacements
+            n_displacements = len(displacements)
+            if n_displacements > 1:
+                for start in range(n_displacements - 1, -1, -1):
+                    for i in range(start+1, len(displacements)):
+                        displacements.append(displacements[start]+displacements[i])
+            images.extend([center_point + d for d in displacements])
+        return np.array(images, dtype=np.float32)
+
+    def search(self, centers, radius):
```

Comment:
I'm not sure you can just change the signature of this method (from `(3)` to `(N, 3)` shaped array).  Some people might have written code around the old behaviour.  Probably better to make a whole new method for multiple searches

Reply:
@richardjgowers `pkdtree` is about one month old and it's been in the develop branch only. It hasn't been released yet

Reply:
... just make sure that these changes go in *before* 0.17.0 is released; once this happens, changing signatures becomes awkward.

Reply:
I added it to the 0.17.0 milestone. 

@jmborr Will you be able to make some progress on this PR? We'd like to get 0.17.0 out soon.